common/logger/src/lib.rs

```diff
@@ -209,6 +210,17 @@ macro_rules! struct_log_enabled {
     };
 }
 
+/// For logging unexpected behavior rather than panicking with `unreachable!`
+#[macro_export]
+macro_rules! unexpected {
+    ($($arg:tt)+) => {
+        let mut entry = $crate::StructuredLogEntry::new_named("unexpected","assertion");
+        entry = entry.data("MUST_FIX", true);
```

Comment:
I'd say either call the category MUST_FIX or let's roll with unexpected instead : o but I don't think we need to have an additional data field

Reply:
Okay, then will roll with unexpected.